### ISSUE TITLE
Post not shown 

### ISSUE THREAD

**Issue by adiell** [airbnb/knowledge-repo#466]
Auto-reviewers: @NiharikaRay @matthewwardrop @earthmancash @danfrankj

Hello, 
I've started a new knowledge repo according to the instructions in the quick-start guide. The server is up and running, but no posts are shown, despite being created. I see no errors of any kind. What can be the problem? How can I debug it?
Thanks,
Adiel

**Comment by adiell** [comment]
Thanks @matthewwardrop . Yes, the code works for me and I do see posts:
```
>>> list(repo.posts())
[<knowledge_repo.post.KnowledgePost object at 0x7f44dc2c3cc0>, <knowledge_repo.post.KnowledgePost object at 0x7f44e3252828>, <knowledge_repo.post.KnowledgePost object at 0x7f44db8025f8>]
```

However when I run the server I don't....

**Comment by matthewwardrop** [comment]
@adiell Sorry! I dropped the ball on this one!

The next step would be to manually start a knowledge repo server from the repository in Python, and see if the posts appear there:
```
repo.get_app().run()
```

If that works, then there is something wrong with the indexing in your configuration, and we can go from there.

**Comment by glema-gpsw** [comment]
@matthewwardrop  when running
repo = KnowledgeRepository.for_uri('<path to checked out out repository; or DB uri>')
I get:
`WARNING:knowledge_repo.repositories.gitrepository:Repository missing configuration file: git:///.knowledge_repo_config.yml`
And then 
```
>>> list(repo.posts())
[<knowledge_repo.post.KnowledgePost at 0x10a634ed0>]
```
Finally:
``` 
repo.get_app().run()
``` 
I don't get any error..but I get nothing in `localhost:7000`
Thanks

**Comment by matthewwardrop** [comment]
@glema-gpsw Thanks for chiming in :). The default port is `5000` when not using the `knowledge_repo` script, since that default is the default offered by Flask, not by the Knowledge Repo.

@adiell @glema-gpsw I can reproduce the posts not appearing, though. It was caused by some recent improvements to the indexing logic, but regressed in non-server deployment instances. I have patched the issue and released v0.8.6.

You could, if you did not want to update, use `app = repo.get_app(); app.start_indexing(); app.run()`; but I recommend updating :).
